### ISSUE TITLE
Bun does not understand "localhost" in fetch 

### ISSUE THREAD

**Issue by pdenapo** [oven-sh/bun#6885]
### What version of Bun is running?

1.0.8+2a405f691e80725fe0b97b93afd3b8cfed13fa5f

### What platform is your computer?

Linux 6.1.0-12-amd64 x86_64 unknown

### What steps can reproduce the bug?

Run the following code that creates a hello-world express sever whith and end
point and fetch it using "localhost".
(I'm using express 4.18.2)

        var express = require("express");
        
        var app = express();
        app.get("/api", function (req, res) {
          res.send("hola");
        });
        var server = app.listen(8000, "127.0.0.1");
        
        const url = "http://localhost:8000";
        
        async function make_request() {
          let response = await fetch(url + "/api");
          const text = await response.text();
          console.log(text);
          server.close();
        }
        
        make_request();

### What is the expected behavior?

Under node it returns "hola". 

### What do you see instead?

ConnectionRefused: Unable to connect. Is the computer able to access the url?
 path: "http://localhost:8000/api"

### Additional information

If you change the line with the server url to 

    const url = "http://127.0.0.1:8000";

everything works in bun as expected!

**Comment by JeffreyATW** [comment]
This doesn't work on Node either. I get this in Node:
```
➜ npx ts-node test.ts
TypeError: fetch failed
    at Object.fetch (node:internal/deps/undici/undici:14152:11)
    at processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: Error: connect ECONNREFUSED ::1:8000
      at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1494:16) {
    errno: -61,
    code: 'ECONNREFUSED',
    syscall: 'connect',
    address: '::1',
    port: 8000
  }
}
```

Works in both Bun and Node if you remove the `"127.0.0.1"` from the `listen` call.

**Comment by pdenapo** [comment]
Strange. Seems to be something related to the network configuration in my machine. I'm using node 20.9.0
 
cat /etc/networks  gives

default		0.0.0.0
loopback	127.0.0.0
link-local	169.254.0.0

However, for some reason bun and node are resolving the host address in a different way!

**Comment by pdenapo** [comment]
My operative system is Debian 12

**Comment by Electroid** [comment]
We will fix this, duplicate of #1425
